tests/NonNegativeFloatString.test.ts

```diff
@@ -0,0 +1,83 @@
+import * as RA from 'fp-ts/ReadonlyArray'
+
+import { tuple } from 'fp-ts/function'
+
+import {
+  Decoder,
+  Eq,
+  Guard,
+  TaskDecoder,
+  toNonNegativeFloat,
+  Type,
+} from '../src/string/NonNegativeFloatString'
+
+import { cat, combineExpected } from '../test-utils'
+
+const validNumbers = [
+  '0',
+  '1',
+  '1.1',
+  `${Math.random() + 1}`,
+  `${Number.MAX_SAFE_INTEGER}`,
+]
```

Comment:
🤔 Here (depending on how the number is parsed) we might want to add other valid inputs, like `'1.1.1.1'`, or `'2......'`

Reply:
There's not an isomorphism between `string <-> number` using `Number` instead of `parseFloat`, but it does eliminate some valid inputs where there's _definitely_ not an isomorphism.

Reply:
It's definitely worth adding these when we add fast-check / retrofit older tests